K8s artifacts generation fails

### Description:
When used deployment.config toml the build failing with an complication issue.
Even after resolving it k8s cluster is not starting giving an error[1]

[1] -
```
java.lang.RuntimeException: java.io.FileNotFoundException: /home/ballerina/conf/micro-gw.conf (No such file or directory)
	at com.moandjiezana.toml.Toml.read(Toml.java:74)
	at org.wso2.micro.gateway.tools.GetConfig.getConfigurations(GetConfig.java:25)
	at org.wso2.micro.gateway.tools.Main.main(Main.java:12)
Caused by: java.io.FileNotFoundException: /home/ballerina/conf/micro-gw.conf (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at com.moandjiezana.toml.Toml.read(Toml.java:72)
	... 2 more

```
### Steps to reproduce:.

### Affected Product Version:
<!-- Members can use Affected/*** labels -->

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---
### Optional Fields
#### Related Issues:
<!-- Any related issues from this/other repositories-->

#### Suggested Labels:
<!--Only to be used by non-members-->

#### Suggested Assignees:
<!--Only to be used by non-members-->

--- a/components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/config/KubernetesDeployment.java
+++ b/components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/config/KubernetesDeployment.java
@@ -40,7 +40,7 @@ public class KubernetesDeployment {
     private String image;
     private String env;
     private String buildImage;
-    private String cmd = "CMD gateway ${APP} --b7a.config.file=conf/micro-gw.conf";
+    private String cmd = "CMD gateway ${APP}";
     private CopyFileConfig copyFiles;
     private String dockerHost;
     private String dockerCertPath;

--- a/components/micro-gateway-cli/src/main/resources/templates/listeners.mustache
+++ b/components/micro-gateway-cli/src/main/resources/templates/listeners.mustache
@@ -5,7 +5,10 @@ import ballerina/io;
 import ballerina/log;
 
 {{#if containerConfig.hasDocker}}import ballerina/docker;{{/if}}
-{{#if containerConfig.kubernetes.kubernetesService.enable}}import ballerina/kubernetes;{{/if}}
+{{#if containerConfig.kubernetes.kubernetesService.enable}}import ballerina/kubernetes;
+{{else if containerConfig.kubernetes.kubernetesServiceHttps.enable}}import ballerina/kubernetes;
+{{else if containerConfig.kubernetes.kubernetesServiceHttp.enable}}import ballerina/kubernetes;{{/if}}
+
 
 //Throttle tier data initiation
 

--- a/distribution/resources/bin/gateway
+++ b/distribution/resources/bin/gateway
@@ -192,7 +192,7 @@ java_cmd_gateway () {
     $JAVA_OPTS \
     -Dmgw-runtime.home=${GW_HOME} \
     -Dballerina.home=${GW_HOME}/runtime \
-    -jar $args --api.usage.data.path=$GW_HOME/api-usage-data --b7a.config.file=$MGW_CONF_DIR/micro-gw.conf 2>&1 | tee -a $GW_HOME/logs/microgateway.log
+    -jar $args --api.usage.data.path=$GW_HOME/api-usage-data --b7a.config.file=$CONFIG_FILE 2>&1 | tee -a $GW_HOME/logs/microgateway.log
 }
 
 java_cmd_gateway_background () {
@@ -205,7 +205,7 @@ java_cmd_gateway_background () {
     $JAVA_OPTS \
     -Dmgw-runtime.home=${GW_HOME} \
     -Dballerina.home=${GW_HOME}/runtime \
-    -jar $args --api.usage.data.path=$GW_HOME/api-usage-data --b7a.config.file=$MGW_CONF_DIR/micro-gw.conf 2>&1 | tee -a $GW_HOME/logs/microgateway.log
+    -jar $args --api.usage.data.path=$GW_HOME/api-usage-data --b7a.config.file=$CONFIG_FILE 2>&1 | tee -a $GW_HOME/logs/microgateway.log
 EOF
 }
 
@@ -251,6 +251,12 @@ fi
 
 OBSERVABILITY_FLAG="false"
 CONFIG_FILE="$MGW_CONF_DIR/micro-gw.conf"
+K8S_MOUNT_PATH="/home/ballerina/conf/ballerina.conf"
+# if externally config map mounted conf found then use that as the config file location. This happens when used with k8s
+if [ -f "$K8S_MOUNT_PATH" ]
+then
+	CONFIG_FILE="$K8S_MOUNT_PATH"
+fi
 CONFIG_OUT_FILE="$GW_HOME/.config"
 
 if [[ $b7a_observability_metrics_enabled == "true" ]]; then

